qualang_tools/addons/calibration/calibrations_test.py

```diff
@@ -0,0 +1,228 @@
+        },
+    },
+    "waveforms": {
+        "zero_wf": {"type": "constant", "sample": 0.0},
+        "const_wf": {"type": "constant", "sample": 0.4},
+        "readout_wf": {"type": "constant", "sample": 0.2},
+        "gauss_wf": {"type": "arbitrary", "samples": gauss(0.25, 0.0, 8, 36)},
+    },
+    "digital_waveforms": {
+        "ON": {"samples": [(1, 0)]},
+    },
+    "integration_weights": {
+        "cosine_weights": {
+            "cosine": [(1.0, 120)],
+            "sine": [(0.0, 120)],
+        },
+        "sine_weights": {
+            "cosine": [(0.0, 120)],
+            "sine": [(1.0, 120)],
+        },
+        "minus_sine_weights": {
+            "cosine": [(0.0, 120)],
+            "sine": [(-1.0, 120)],
+        },
+    },
+    "mixers": {
+        "mixer_qubit": [
+            {
+                "intermediate_frequency": 50e6,
+                "lo_frequency": 5e9,
+                "correction": IQ_imbalance(0.0, 0.0),
+            }
+        ],
+        "mixer_resonator": [
+            {
+                "intermediate_frequency": 50e6,
+                "lo_frequency": 5e9,
+                "correction": IQ_imbalance(0.0, 0.0),
+            }
+        ],
+    },
+}
+
+# Relevant configuration parameters
+#############################################################
+resonator_element = "resonator"
+resonator_operation = "readout"
+qubit_element = "qubit"
+qubit_operation = "gaussian"
+flux_line_element = None
+flux_line_operation = None
+int_w = ["cos", "sin", "minus_sin", "cos"]
+outs = ["out1", "out2"]
+#############################################################
+
+# Initialize the calibration API with the relevant configuration parameters
+my_calib = QUA_calibrations(
+    configuration=config,
+    readout=(resonator_element, resonator_operation),
+    qubit=(qubit_element, qubit_operation),
+    integration_weights=int_w,
+    outputs=outs,
+)
+
+n_avg = 1000
+# Resonator spectroscopy scan
+scans = [
+    ("frequency", np.arange(100e6, 300e6, step=0.1e6)),
+]
+# my_calib.set_resonator_spectroscopy(
+#     scan_variables=scans, iterations=n_avg, cooldown_time=0
+# )
+
+# Rabi scan
+scans = [
+    ("duration", np.arange(40, 4000, step=100)),
+    ("amplitude", np.linspace(0.5, 1.99, num=51)),
+]
+# my_calib.set_rabi(scan_variables=scans, iterations=n_avg, cooldown_time=0)
+
+# T1 scan
+scans = [
+    ("duration", np.arange(5, 4000, step=10)),
+]
+my_calib.set_T1(scan_variables=scans, iterations=n_avg, cooldown_time=0)
+
+
+# Ramsey scan
+scans = [
+    ("frequency", np.linspace(100e6, 200e6, num=101)),
+    ("duration", np.arange(40, 4000, step=10)),
+]
+my_calib.set_ramsey(
+    scan_variables=scans, iterations=n_avg, cooldown_time=0, idle_time=10
+)
+
+# Raw traces
+# my_calib.set_raw_traces(iterations=n_avg, cooldown_time=0)
+
+# Time of flight
+# my_calib.set_time_of_flight(iterations=n_avg, cooldown_time=0)
+
+# Open QM
+qmm = QuantumMachinesManager(host="172.16.2.103", port=80)
```

Comment:
in the test you need to be connected thru the VPN?

Reply:
This is how I tested it yes